File: code/stable/nopcm/src/csharp/Control.cs
/** \file Control.cs
    \author Thulasi Jegatheesan
    \brief Controls the flow of the program
*/
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;

public class Control {
    
    /** \brief Controls the flow of the program
        \param args List of command-line arguments
    */
    public static void Main(string[] args) {
        string filename = args[0];
        double A_C;
        double C_W;
        double h_C;
        double T_init;
        double t_final;
        double L;
        double T_C;
        double t_step;
        double rho_W;
        double D;
        double A_tol;
        double R_tol;
        double T_W;
        double E_W;
        InputParameters.get_input(filename, out A_C, out C_W, out h_C, out T_init, out t_final, out L, out T_C, out t_step, out rho_W, out D, out A_tol, out R_tol, out T_W, out E_W);
        InputParameters.input_constraints(A_C, C_W, h_C, T_init, t_final, L, T_C, t_step, rho_W, D, T_W, E_W);
        OutputFormat.write_output(T_W, E_W);
    }
}

File: code/stable/PDController/src/python/Control.py
## \file Control.py
# \author Naveen Ganesh Muralidharan
# \brief Controls the flow of the program
import sys

import Calculations
import InputParameters
import OutputFormat

filename = sys.argv[1]
r_t, K_d, K_p, t_step, t_sim = InputParameters.get_input(filename)
InputParameters.input_constraints(r_t, K_d, K_p, t_step, t_sim)
y_t = Calculations.func_y_t(K_d, K_p, r_t, t_sim, t_step)
OutputFormat.write_output(y_t)
